hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSEditLogAsync.java

```diff
@@ -227,8 +230,33 @@ private Edit dequeueEdit() throws InterruptedException {
     return syncWaitQ.isEmpty() ? editPendingQ.take() : editPendingQ.poll();
   }
 
+  private static class EditSyncEx {
+    final Edit edit;
+    final RuntimeException ex;
+    EditSyncEx(Edit edit, RuntimeException ex) {
+      this.edit = edit;
+      this.ex = ex;
+    }
+  }
+
+  private class LogSyncNotifyThread extends Thread {
+    volatile boolean stopped = false;
+
+    @Override
+    public void run() {
+      try {
+        while (!stopped) {
+          EditSyncEx editSyncEx = logSyncNotifyQ.take();
+          editSyncEx.edit.logSyncNotify(editSyncEx.ex);
+        }
+      } catch(InterruptedException ie) {} // just swallow it
+    }
+  }
+
   @Override
   public void run() {
+    final LogSyncNotifyThread logSyncNotifyThread = new LogSyncNotifyThread();
+    logSyncNotifyThread.start();
```

Comment:
single thread to push the result back to client here, I think it is safe to trigger multi-thread to do that. right?